src/vs/platform/quickinput/browser/quickPickPin.ts

```diff
@@ -29,10 +29,6 @@ export async function showWithPinnedItems(storageService: IStorageService, stora
 			quickPick.items = quickPick.value ? itemsWithoutPinned : itemsWithPinned;
 		}
 	});
```

Comment:
the idea here was to not show pinned items when searching. this breaks things though. @TylerLeonhardt is there a better way to accomplish this?

Reply:
not really... you have to setup the event listener to know when you go from no `value` to `value`... I guess one thing you could do is detect when the value goes from `""` to `"a"`, set items, then from `"a"` to `"ab"` you just no-op because you already filtered out the pinned items.

Reply:
I think it's not worth filtering the pinned ones out given the cost (the items erroneously get reset to be the top level ones instead of the current second level ones)

Reply:
idk I'd need to play around with it/see a gif of the change.

Are you sorting the results so the pinned ones get mixed in with the others?

Reply:
https://user-images.githubusercontent.com/29464607/199295564-f5b06418-7c22-4165-ab64-8da2543f77de.mov

Reply:
and here's the bug https://github.com/microsoft/vscode/issues/164629#issuecomment-1298858479